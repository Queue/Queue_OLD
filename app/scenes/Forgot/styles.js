//
// SignIn styles.js

import { StyleSheet, Dimensions } from 'react-native';
import Colors from '../../lib/colors';

let { width, height }  = Dimensions.get('window');

export default styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: width,
    height: height,
    backgroundColor: Colors.primaryBackground
  },
  wrapper: {
    width: 300
  },
  brand: {
    fontSize: 142,
    fontFamily: 'airstream',
    marginBottom: -40,
    color: Colors.primaryForeground,
    textAlign: 'center'
  }
});
