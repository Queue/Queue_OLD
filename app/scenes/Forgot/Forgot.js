//
// Forgot Scene

import React, { Component } from 'react';
import styles from './styles';
import {View,  Text} from 'react-native';
import {
  PrimaryButton,
  EmailField,
  PasswordField,
  TextButton
} from '../../components';
import KeyboardSpacer from 'react-native-keyboard-spacer';
import Data from '../../lib/data';
import Colors from '../../lib/colors';
import { Actions } from 'react-native-router-flux'
import Common from '../../lib/common';

export default class Forgot extends Component {
  constructor(props) {
    super(props);

    this.state = {
      emailText: ''
    };
  }

  resetPassword() {
    let email = this.state.emailText;

    if (email !== '' && Common.validateEmail(email)) {
      Data.Auth.resetPassEmail(email).then(() => {
        Common.log('Success', 'Password reset email sent.');
        Common.dismissKeyboard();
        Actions.SignInRoute();
      }, (error) => {
        Common.error('Error', 'Email is not registered.');
      });
    } else {
      Common.error('Error', 'Enter your email');
    }
  }

  render() {
    let email = this.state.emailText;

    return (
      <View style = {styles.container}>
        <View style = {styles.wrapper}>
          <Text
            style = {styles.brand}>
            Queue
          </Text>
          <EmailField
            placeholder = {'Email'}
            focus={this.state.focus}
            change = {(text) => this.setState({emailText: text})}
            val = {email}
          />
          <PrimaryButton
            name = {'Reset Password'}
            press = {this.resetPassword.bind(this)}
          />
          <TextButton
            size = {16}
            text = {'Backâ†µ '}
            press = {Actions.SignInRoute}
          />
          <KeyboardSpacer />
        </View>
      </View>
    );
  }

}
