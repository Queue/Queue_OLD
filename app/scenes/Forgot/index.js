//
// Forgot index.js

import Forgot from './Forgot';
export default Forgot;
