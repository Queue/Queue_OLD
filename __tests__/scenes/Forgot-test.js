//
// Forgot scene test

import React from 'react';
import Forgot from '../../app/scenes/Forgot';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Forgot />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
