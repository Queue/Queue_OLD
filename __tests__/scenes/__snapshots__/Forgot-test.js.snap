exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eaeaea",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "width": 300,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#4AC948",
          "fontFamily": "airstream",
          "fontSize": 142,
          "marginBottom": -40,
          "textAlign": "center",
        }
      }>
      Queue
    </Text>
    <TextInput
      autoCapitalize="none"
      autoFocus={undefined}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 2,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          Object {
            "color": "#4AC948",
          },
        ]
      }
      value="" />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#4AC948",
          "borderRadius": 4,
          "borderWidth": 1,
          "margin": 15,
          "marginTop": 20,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 8,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Array [
              Object {
                "fontSize": 20,
                "fontWeight": "500",
              },
              Object {
                "color": "#4AC948",
              },
            ],
            null,
          ]
        }>
        Reset Password
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#4AC948",
              "fontSize": 14,
              "padding": 5,
              "paddingBottom": 15,
              "textAlign": "center",
            },
            Object {
              "fontSize": 16,
            },
          ]
        }>
        Backâ†µ 
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
          },
          Object {
            "height": 0,
          },
          undefined,
        ]
      } />
  </View>
</View>
`;
